api/src/main/java/org/ehcache/expiry/Expiry.java

```diff
@@ -56,6 +58,6 @@
    * @param newValue the new value of the entry
    * @return the updated expiration value for the given entry. A {@code null} return value indicates "no change" to the expiration time
    */
-  Duration getExpiryForUpdate(K key, V oldValue, V newValue);
+  Duration getExpiryForUpdate(K key, Invoker<V> oldValue, V newValue);
```

Comment:
I like these changes, they seem to make the API more honest.  Retrieving the values is something being done that may otherwise not be done... so it has a cost.  Associating a cost with an action is a nice way of communicating to the user that what they are doing might have a performance impact.